packages/protoc-gen-es/src/transpile.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { TSFile } from "@bufbuild/protoplugin/ecmascript";
+import * as ts from "typescript";
+import {
+  createDefaultMapFromNodeModules,
+  createSystem,
+  createVirtualCompilerHost,
+} from "@typescript/vfs";
+
+function createTranspiler(options: ts.CompilerOptions, files: TSFile[]) {
+  const fsMap = createDefaultMapFromNodeModules({
+    target: ts.ScriptTarget.ES2015,
```

Comment:
This `ScriptTarget` value may need to change (or we can just use the `options` passed in since this accepts `CompilerOptions`.  We just need to solidify what are default options are.

Reply:
Let's use ES2017 as the script target, following https://github.com/bufbuild/connect-web#supported-browsers

Reply:
Oh, I see we're setting module CommonJS in the line below. We definitely need to use ECMAScript module imports here, and need to make sure we include the `.js` extension. I think this is worth a test case.